From b066b235daae3d072a35d7a56ab6e65715065a9a Mon Sep 17 00:00:00 2001
From: Josh Poimboeuf <jpoimboe@redhat.com>
Date: Thu, 4 Sep 2014 18:54:37 -0400
Subject: [kernel] ftrace: Do no disable function tracing on enabling function tracing

Message-id: <9b767a01b30592e07326a27f3aac0c88d78c1963.1409856554.git.jpoimboe@redhat.com>
Patchwork-id: 91254
O-Subject: [RHEL7.1 PATCH 18/23] ftrace: Do no disable function tracing on enabling function tracing
Bugzilla: 1113829
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1113829

commit 1820122a76c6d64adc6e2a7ff438029ffb8d7cb4
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date:   Wed Jun 25 11:28:20 2014 -0400

    ftrace: Do no disable function tracing on enabling function tracing

    When function tracing is being updated function_trace_stop is set to
    keep from tracing the updates. This was fine when function tracing
    was done from stop machine. But it is no longer done that way and
    this can cause real tracing to be missed.

    Remove it.

    Reviewed-by: Masami Hiramatsu <masami.hiramatsu.pt@hitachi.com>
    Signed-off-by: Steven Rostedt <rostedt@goodmis.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/trace/ftrace.c b/kernel/trace/ftrace.c
index b4e1f39..8db794a 100644
--- a/kernel/trace/ftrace.c
+++ b/kernel/trace/ftrace.c
@@ -2002,11 +2002,6 @@ static void ftrace_run_update_code(int command)
  FTRACE_WARN_ON(ret);
  if (ret)
   return;
- /*
-  * Do not call function tracer while we update the code.
-  * We are in stop machine.
-  */
- function_trace_stop++;
 
  /*
   * By default we use stop_machine() to modify the code.
@@ -2016,8 +2011,6 @@ static void ftrace_run_update_code(int command)
   */
  arch_ftrace_update_code(command);
 
- function_trace_stop--;
-
  ret = ftrace_arch_code_modify_post_process();
  FTRACE_WARN_ON(ret);
 }
-- 
1.7.1